
This patch file includes the changes required to add
compat-wireless support. Besides all the patches in this
directory we have our own compat-2.6.*.[ch] files which
help backport more.

We could optimize this to not compile objects for
kernels >= your own kernel but that may mean
some nasty check here, feel free to add it if you
figure out how.

Lastly, the cfg80211.h change is what allows us to
get all compat-2.6.*.h files included. Its our only
point of entry through compat code.

If your driver is not a cfg80211 driver you must
ensure you include compat.h then. I think we
added compat.h also for ssb, I forget.

--- a/net/wireless/Makefile	2009-08-07 12:27:50.836497001 -0700
+++ b/net/wireless/Makefile	2009-08-07 12:27:51.952497240 -0700
@@ -11,3 +11,11 @@
 cfg80211-$(CONFIG_WIRELESS_EXT) += wext-compat.o wext-sme.o
 
 ccflags-y += -D__CHECK_ENDIAN__
+# Compat-wireless kernel compatibility code
+cfg80211-y += compat-2.6.27.o
+cfg80211-y += compat-2.6.28.o
+cfg80211-y += compat-2.6.29.o
+cfg80211-y += compat-2.6.30.o
+cfg80211-y += compat-2.6.31.o
+cfg80211-y += compat-2.6.32.o
+
--- a/include/net/cfg80211.h
+++ b/include/net/cfg80211.h
@@ -23,6 +23,7 @@
 /* remove once we remove the wext stuff */
 #include <net/iw_handler.h>
 #include <linux/wireless.h>
+#include <net/compat.h>
 
 
 /*
